packages/material-ui/src/Slider/Slider.test.js

```diff
@@ -970,4 +970,22 @@ describe('<Slider />', () => {
       expect(getByTestId('value-label')).to.have.text('1010');
     });
   });
+
+  it('should not leak the event.target on touch events', () => {
+    const touchStart = spy();
+    function Test() {
+      React.useEffect(() => {
+        document.addEventListener('touchstart', touchStart);
+        return () => {
+          document.removeEventListener('touchstart', touchStart);
+        };
+      });
+
+      return <Slider value={0} onChange={() => {}} />;
+    }
+
+    const { container } = render(<Test />);
+    fireEvent.touchStart(container.firstChild, createTouches([{ identifier: 1 }]));
+    expect(touchStart.args[0][0].target.nodeType).to.equal(1);
+  });
```

Comment:
I don't understand how the test description relates to what's being done in the test. When we talk about "leaks" we usually mean that something is still in memory even though we no longer need it. But that doesn't seem to be the case here.

This doesn't seem to be a "leak" issue but just fixing how `event.target` is defined. Before we  added a custom object (arguably `target` was just wrong given that you would need to access the actual element with `event.target.target`). But I'm not really understanding what changed. This seems like very smelly code to me.

I need to look into the code a bit more to understand what `target` actually refers to.

Reply:
Maybe I didn't use the right term. How about "global side effect"?

Reply:
I have added the support for `event.target.target` to anticipate the need for somebody doing something advanced. We could be lazy and remove it.

Reply:
I replaced "leak" with "override" and added a test for `onMouseDown`. I don't think the assumption in

https://github.com/mui-org/material-ui/blob/a15b07dc2d5743ad24bd1ecb94ede30e163720fd/packages/material-ui-unstyled/src/SliderUnstyled/SliderUnstyled.js#L240

is correct.